docs/docs/concepts/foundation/communication/public_private_calls/slow_updates_tree.md

```diff
@@ -0,0 +1,108 @@
+---
+title: Privately access Historical Public Data
+---
+
+In Aztec, private and public execution environments are completely separate and operate with distinct state management. It is not possible for private functions to reliably access the most recent public data public state - only sequencers can do that.
+
+But, what about historical public data (or public data that changes infrequently)? Through a **slow updates tree**, private functions can access historical public state. Please note that we are still experimenting with this feature.
+
+On this page you will learn:
+
+1. Why a slow updates tree exists & use cases
+2. How it works
+3. How it can be used to access historical public data
+4. Limitations
+
+## The need for a slow updates tree
+
+This structure was created specifically to privately & publicly access historical public data. It should be used to store public data that doesn't change often.
+
+- Access historical public data from a private function
+- Access historical public data from a public function
+- Update public data (that does not need updated often) from public and private functions 
+
+This data structure is ideal for these use cases:
+
+- Address Registry: Enabling contracts to interact with other contracts more easily requires address storage accessible in both public and private executions. This can be particularly helpful in things such as proxy contracts.
+- Access Control: Managing privileges in contracts, such as a token contract owner’s ability to mint new tokens, is streamlined when control information is shared between public and private executions. This might include things like blacklists and whitelists.
+
+## How it works
+
+We developed the Slow Updates Tree to help balance public and private execution in a blockchain context. Earlier systems typically used either fully public or entirely private state trees.
+
+**First iteration: Shared State Tree** 
+
+Using a shared state tree, it is possible to privately access historical public data by providing a membership proof to show that a value is indeed part of a commitment, and then check that the commitment matches the one stored in the state, as shown below.
+
+```mermaid
+graph TD;
+    Cm[Commmitment1] --> Vm1[Value1]
+    Cm --> Vm2[Value2]
+    Cm --> Vmn[Value3]
+    Vm1 --> Mp1[Membership Proof for Value1]
+    Mp1 --> St[Commitment in State]
+    St --> Cm
+```
+
+Note: a *commitment* refers to a cryptographic assurance that a specific set of data is included in the tree. In the current design, we are using Merkle trees to commit to the values.
+
+However, these would be contract-specific trees, so we will be leaking the contract address of the contract doing a lookup. 
+
+**Second iteration: Multi-level Shared State Tree**
```

Comment:
This is the 2nd iteration but rather the final one and doesn't exist yet. Best to delete.